pkg/config/config.go

```diff
@@ -61,27 +61,23 @@ type Config struct {
 	EvalContext *hcl.EvalContext
 }
 
-func loadLokocfgPaths(configPath string) ([]string, error) {
-	isDir, err := util.PathIsDir(configPath)
-	if err != nil {
-		return nil, fmt.Errorf("failed to stat config path %q: %w", configPath, err)
-	}
-	var lokocfgPaths []string
-	if isDir {
-		globPattern := filepath.Join(configPath, "*.lokocfg")
-		configFiles, err := filepath.Glob(globPattern)
-		if err != nil {
-			return nil, fmt.Errorf("bad filepath glob pattern %q: %w", globPattern, err)
+// Component finds a component named name and returns its configuration as a pointer to an
+// hcl.Body. If the component isn't found, nil is returned.
+func (c *Config) Component(name string) *hcl.Body {
+	for _, component := range c.RootConfig.Components {
+		if name == component.Name {
+			return &component.Config
 		}
-		lokocfgPaths = append(lokocfgPaths, configFiles...)
-	} else {
-		lokocfgPaths = append(lokocfgPaths, configPath)
 	}
-	return lokocfgPaths, nil
+
+	return nil
 }
 
-func LoadConfig(lokocfgPath, lokocfgVarsPath string) (*Config, hcl.Diagnostics) {
-	lokocfgPaths, err := loadLokocfgPaths(lokocfgPath)
+// New reads HCL files at path and HCL variable files at varPath, constructs a Config and returns a
+// pointer to it.
+//nolint:funlen
+func New(path, varPath string) (*Config, hcl.Diagnostics) {
```

Comment:
Again, `LoadConfig` seems to me like the right name for this function. `New` is usually used as a constructor to get new instance of the object, however this function executes specific action, which is loading the configuration using given paths. We could do things llike `config.Load()` or `config.LoadHCL()`, but just `New` seems more ambiguous to me.

Reply:
I don't think `LoadConfig` is a good name, because:

- "Load" is a very ambiguous term IMO.
- "Config" is stuttering (`config.LoadConfig()` could be e.g. `config.Load()`).

How about `config.ReadHCL()`? IMO "read" makes it clearer we are reading files from disk.

Reply:
I've changed to `config.ReadHCL()` to save time. I can change again if you still have concerns.